dask/dataframe/io/parquet/arrow.py

```diff
@@ -625,6 +654,181 @@ def _construct_parts(
+        base, fns = _analyze_paths(paths, fs)
+        paths = fs.sep.join([base, fns[0]])
+
+        meta_path = fs.sep.join([paths, "_metadata"])
+        if fs.exists(meta_path):
+            # Use _metadata file if available
+            ds = pa_ds.parquet_dataset(
+                meta_path, partitioning=dataset_kwargs.get("partitioning", "hive")
+            )
+            if gather_statistics is None:
+                gather_statistics = True
+    elif len(paths) > 1:
+        base, fns = _analyze_paths(paths, fs)
+        meta_path = fs.sep.join([base, "_metadata"])
+        if "_metadata" in fns:
+            # Use _metadata file if available
+            # Pyarrow cannot handle "_metadata" when `paths` is a list
+            paths.remove(meta_path)
+            fns.remove("_metadata")
+            ds = pa_ds.parquet_dataset(
+                meta_path, partitioning=dataset_kwargs.get("partitioning", "hive")
+            )
+            if gather_statistics is None:
+                gather_statistics = True
+
+    if ds is None:
+        ds = pa_ds.dataset(
+            paths,
+            format="parquet",
+            partitioning=dataset_kwargs.get(
+                "partitioning", "hive"
+            ),  # Assume "hive" by default
+        )
+
+    if split_row_groups is None and gather_statistics is not False:
+        split_row_groups = True
+
+    # Generate list of row-group fragments and call it `metadata`
+    metadata, partition_info = _collect_pyarrow_dataset_frags(ds, filters)
+    schema = ds.schema
+    base = ""
+
+    return (schema, metadata, base, partition_info, split_row_groups, gather_statistics)
+
+
+def _process_metadata_pyarrow_dataset(
+    metadata, single_rg_parts, gather_statistics, stat_col_indices, no_filters
+):
+
+    #
+    # TODO: Use `no_filters` to perform "early bail" opt.
+    #
+
+    # Get the number of row groups per file
+    file_row_groups = defaultdict(list)
+    file_row_group_stats = defaultdict(list)
+    file_row_group_column_stats = defaultdict(list)
+    for rg_frag in metadata:
+        row_group = rg_frag.row_groups[0]
+        fpath = rg_frag.path
+        file_row_groups[fpath].append(rg_frag)
+        statistics = row_group.statistics
+        if gather_statistics:
+            if single_rg_parts:
+                s = {
+                    "file_path_0": fpath,
+                    "num-rows": row_group.num_rows,
+                    "total_byte_size": row_group.total_byte_size,
+                    "columns": [],
+                }
+            else:
+                s = {
+                    "num-rows": row_group.num_rows,
+                    "total_byte_size": row_group.total_byte_size,
+                }
+            cstats = []
+            for name, i in stat_col_indices.items():
+                if name in statistics:
+                    cmin = statistics[name]["min"]
+                    cmax = statistics[name]["max"]
+                    cnull = 0  # Not yet available
```

Comment:
Does dask use this? (if so, should be easy to add in pyarrow, can open a JIRA then)